src/include/transaction/transaction_context.h

```diff
@@ -129,5 +147,8 @@ class TransactionContext {
   // log manager will set this to be true when log records are processed (not necessarily flushed, but will not be read
   // again in the future), so it can be garbage-collected safely.
   bool log_processed_ = false;
+
+  // Functions to be invoked on rollback
+  std::map<transaction::rollback_fn, void *> rollback_functions_;
```

Comment:
When @jrolli mentioned this to me the other day it sounded like a stack was what we were going to use. Do we need ordered and random element access that a map provides?

Reply:
We don't need random access. AFAIK all constraint checks are commutative, so the order does not matter.

Reply:
Suggest replacing with std::forward_list then.

Reply:
Alright, I'll go ahead and do that